addons/barcodes/static/tests/barcode_tests.js

```diff
@@ -67,7 +67,11 @@ QUnit.test('Button with barcode_trigger', function (assert) {
         res_id: 2,
         services: [NotificationService.extend({
             notify: function (params) {
-                assert.step(params.type);
+                // an error message is displayed (with methode do_warn()), then params.type is equal to "warning"
+                // if the message is displayed with do_notify(), then params.type is undefined.
+                if (params.type) {
+                    assert.step(params.type);                    
+                }
```

Comment:
i don't like it ; the issue is that there's no "type" in the params when the call is "do_notify" (there's a type in"do_warn")

i think you should either remove the assert altogether or add an explicit type in "do_notify"